src/main/java/org/vanilladb/core/util/Timer.java

```diff
@@ -107,6 +138,11 @@ public long getComponentCount(Object component) {
 	public List<Object> getComponents() {
 		return new LinkedList<Object>(componenents);
 	}
+	
+	// Where tx started doesn't have timer, so we need to pass the start time into SPTask
+	public void setStartExecution(long start) {

```

Comment:
Please change the signature to `public void setStartExecutionTime(long startTime)`.